js/copy-service/Parser/Parser.js

```diff
@@ -414,10 +431,16 @@ class Parser {
 
     else if (token.type === this.TOKENS.REF_START) {
       const textParsed = this._getTextToken(tokensToParse);
+
+      if (_.startsWith(textParsed.text, '.')) {
+        const parentKey = this._getParentKey(key);
+        textParsed.text = parentKey.concat(textParsed.text);
+      }
```

Comment:
A little bit of a hack, could probably belong in `_getTextToken`?

Reply:
We only need to do this in the context of the text token being a reference to a copy key. To that end, we may want a `_getReferenceKeyToken` method that handles calling `_getRelativeKey`.